pkg/config/nerdctl_config_applier.go

```diff
@@ -91,11 +91,18 @@ func addLineToBashrc(fs afero.Fs, profileFilePath string, profStr string, cmd st
 func updateEnvironment(fs afero.Fs, fc *Finch, finchDir, homeDir, limaVMHomeDir string) error {
 	cmdArr := []string{
 		`export DOCKER_CONFIG="$FINCH_DIR"`,
-		"[ -L /usr/local/bin/docker-credential-ecr-login ] " +
-			`|| sudo ln -s "$FINCH_DIR"/cred-helpers/docker-credential-ecr-login /usr/local/bin/`,
 		`[ -L /root/.aws ] || sudo ln -fs "$AWS_DIR" /root/.aws`,
 	}
 
+	for _, credHelper := range fc.CredsHelpers {
+		cmdArr = append(cmdArr,
+			fmt.Sprintf(
+				`[ -L /usr/local/bin/docker-credential-%s ] || sudo ln -s "$FINCH_DIR"/cred-helpers/docker-credential-%s /usr/local/bin/`,
```

Comment:
Can we add a explicit error/warning here if the source binary is not present in the cred-helpers dir?

Reply:
Addressed by https://github.com/runfinch/finch/pull/1265/commits/360785dbbf1bf430e224ae2da2e298b10e85ce37. As symbolic link would be successful even if the source did not exist, I added logic to first check if the source exists before checking if the symbolic link exists.

Reply:
This is pretty much just an info/warning message though, right? Since we aren't checking if there's an error and then returning it back in Go. That's probably fine for now

Reply:
@pendo324 pinged me offline because I missed this comment. For completeness, yes it is best effort to inform users the configuration would not work as expected. The issue is this logic is added to the user's guest machine bashrc and is invoked on each command. Note: container finch commands shell (SSH) into the guest machine and execute nerdctl. The message would not be propagated up through Finch CLI though.